### DISCUSSION TITLE
Openapi spec not updated when overriding dependency with signature change

### DISCUSSION THREAD

**Discussion by benbovy** [fastapi/fastapi#9080] [Questions]
### First check

* [x] I added a very descriptive title to this issue.
* [x] I used the GitHub search to find a similar issue and didn't find it.
* [x] I searched the FastAPI documentation, with the integrated search.
* [x] I already searched in Google "How to X in FastAPI" and didn't find any information.
* [x] I already read and followed all the tutorial in the docs and didn't find an answer.
* [ ] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
* [ ] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
* [ ] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).
* [x] After submitting this, I commit to one of:
    * Read open issues with questions until I find 2 issues where I can help someone and add a comment to help there.
    * I already hit the "watch" button in this repository to receive notifications and I commit to help at least 2 people that ask questions in the future.
    * Implement a Pull Request for a confirmed bug.

### Example

Here's a self-contained, [minimal, reproducible, example](https://stackoverflow.com/help/minimal-reproducible-example) with my use case:

```Python
from fastapi import Depends, FastAPI


ressources = {'foo': [1, 2, 3], 'bar': [4, 5, 6]}


def get_ressource(ressource_id: str):
    return ressources[ressource_id]


app = FastAPI()


@app.get("/")
def get_value(ressource: list = Depends(get_ressource)):
    return ressource


# override `get_ressource` dependency with func signature change

def get_foo():
    return ressources['foo']

app.dependency_overrides[get_ressource] = get_foo
```

### Description

In the example above, the API docs entry for path "/" still shows `ressource_id` as a required parameter, although it is not present in `get_foo`. When testing this API endpoint, it still works fine with any value given for `ressource_id` (it is just ignored).

So my question is: does FastAPI supports overriding dependencies with a change in function signature? Is the unchanged openapi specs a bug or a design choice/limitation?

For more context, a real-world use case is discussed here: https://github.com/xarray-contrib/xpublish/issues/25#issuecomment-645457226. This use case shows that overriding dependencies are incredibly useful, not only for testing.

### Environment

* OS: macOS
* FastAPI version: ~~0.55.0~~ EDIT: 0.58.0
* Python version: 3.8

**Comment by YuriiMotov** [comment]
Duplicate: https://github.com/fastapi/fastapi/discussions/8392

PR: https://github.com/fastapi/fastapi/pull/5452 (rejected)

Explanation: https://github.com/fastapi/fastapi/pull/5452#issuecomment-2979998948
